### DISCUSSION TITLE
SetColumn: labels and colors

### DISCUSSION THREAD

**Discussion by keckelt** [lineupjs/lineupjs#447] [Q&A] [answered]
Hi!

How can I change labels and color of the categories in a SetColumn?

For categorical columns i use the `categories` function of the CateogoricalColumnBuilder, but I think there is no such builder for the SetColumns.

Her's what I'm doing:

```js
const data = [
  {tissuename: 'TCGA1', cohort: ['1', '2', '3']},
  {tissuename: 'TCGA2', cohort: ['1']},
  {tissuename: 'TCGA3', cohort: ['1', '3']}
];

const setColumnDesc = {type: 'set', label: 'Set', column: 'cohort', categories: ['1', '2', '3']};

const builder = LineUpJS.builder(data)
  .column(LineUpJS.buildStringColumn('tissuename'))
  .column(setColumnDesc)
  .build(document.querySelector('#lineup'));
```
Codepen: https://codepen.io/keckelt/pen/JjWmjMJ

I would like to specify the color (e.g., 1=🟡,2=🟠,3=🔴) and label (e.g., _Lung Cancer, Colon Cancer, Breast Cancer_) of each category.

**Comment by sgratzl** [comment, marked as answer]
The `CategoricalColumnBuilder` has a `.asSet()` method converting a categorical definition in a set definition. 

Then you can do just: 

https://codepen.io/sgratzl/pen/QWpYNby?editors=0010

![image](https://user-images.githubusercontent.com/4129778/122215122-29be5100-ceab-11eb-98c5-28070431e4a4.png)
